lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -381,9 +459,38 @@ internal int NumberOfAppErrors()
         /// Checks if all the tasks are in final states
         /// </summary>
         /// <returns></returns>
-        internal bool AllInFinalState()
+        internal bool AreAllTasksInFinalState()
+        {
+            var notInFinalState = _tasks.Where(t => !t.Value.TaskState.IsFinalState()).Take(5).ToList();
```

Comment:
I understand that number 5 or 1 or 10 doe snot make a difference here. However, in my opinion this is more for debugging purpose and should be removed.